File: integrations/claude-agent-sdk/typescript/examples/agentic_chat_reasoning.ts
/**
 * Agentic chat with reasoning example.
 *
 * This example shows how to create an agentic chat adapter with
 * extended thinking/reasoning enabled using Claude Haiku 4.5.
 */

import { ClaudeAgentAdapter } from "@ag-ui/claude-agent-sdk";
import { DEFAULT_DISALLOWED_TOOLS } from "./constants";

/**
 * Create adapter for agentic chat with reasoning enabled.
 *
 * Uses maxThinkingTokens to enable extended thinking so Claude
 * can reason through complex questions step by step.
 */
export function createAgenticChatReasoningAdapter(): ClaudeAgentAdapter {
  return new ClaudeAgentAdapter({
    agentId: "agentic_chat_reasoning",
    description: "Chat assistant with extended thinking/reasoning",
    model: "sonnet",
    systemPrompt: "You are a helpful assistant. Think step by step when answering questions.",
    thinking: { type: "enabled", budgetTokens: 10000 },
    includePartialMessages: true,
    disallowedTools: [...DEFAULT_DISALLOWED_TOOLS],
  });
}

File: integrations/claude-agent-sdk/python/examples/agents/agentic_chat_reasoning.py
"""
Agentic chat with reasoning agent configuration.

This module provides a factory function for creating an agentic chat adapter
with extended thinking/reasoning enabled. Uses Claude Haiku 4.5 with adaptive
thinking for cost-effective reasoning demonstrations.
"""

from ag_ui_claude_sdk import ClaudeAgentAdapter
from .constants import DEFAULT_DISALLOWED_TOOLS


def create_agentic_chat_reasoning_adapter() -> ClaudeAgentAdapter:
    """Create adapter for agentic chat with reasoning enabled."""
    return ClaudeAgentAdapter(
        name="agentic_chat_reasoning",
        description="Chat assistant with extended thinking/reasoning",
        options={
            "model": "sonnet",
            "system_prompt": "You are a helpful assistant. Think step by step when answering questions.",
            "thinking": {"type": "enabled", "budget_tokens": 10000},
            "disallowed_tools": list(DEFAULT_DISALLOWED_TOOLS),
        }
    )
